iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ClientOptions.java

```diff
@@ -136,4 +136,12 @@
     @Getter
     @Builder.Default
     private final int receiveInterval = RECEIVE_PERIOD_MILLIS;
+
+    /**
+     * This option specifies the type of gateway to which the device/module client is connecting while
+     * specifying the property of GatewayHostName. By default, the value is EDGE.
+     */
+    @Getter
+    @Builder.Default
+    private final GatewayType gatewayType = GatewayType.EDGE;
```

Comment:
We'll eventually want to add in some additional comments that call out the operations that are unsupported over E4K gateway type.

Reply:
Should the default value here be something like ```NONE```? User's may be confused if they have to pick between E4K and EDGE when they don't want to use a gateway at all

Reply:
Yeah, you are right.